app/client/components/payment/update/PaymentDetailUpdate.tsx

```diff
@@ -214,6 +215,14 @@ const PaymentDetailUpdate = (props: WithProductType<ProductType>) => {
 
 	const currentPaymentMethod = subscriptionToPaymentMethod(productDetail);
 
+	const directDebitIsAllowed =
+		currentPaymentMethod == PaymentMethod.dd || (
```

Comment:
tiny suggestion:

`currentPaymentMethod === PaymentMethod.dd`

sneaky strict equality check in there